src/app.js

```diff
@@ -81,134 +33,58 @@ var run = function() {
-
-		if (typeof chrome === 'undefined' || !chrome || !chrome.storage || !chrome.storage.sync) return;
-
-		var obj = {};
-		var done = false;
-		var count = 0;
-		var dont = 0;
-
-		function check(key){
-			chrome.storage.sync.get( key, function(resp){
-				console.log("comparing to cloud", key, resp);
-				if (typeof resp[key] != 'undefined' && obj[key] !== resp[key] && confirm("Overwrite '" + fileNameFromKey(key) + "'? Click Ok to overwrite local file with file from cloud. Cancel will push your local file to the cloud.")) {
-					console.log("Overwriting", key );
-					localStorage.setItem( key, resp[key] );
-					updateFiles();
-				} else {
-					console.log( "add to obj", obj, key);
-					obj[key] = localStorage[key];
-				}
-				done++;
-				if (done >= count) chrome.storage.sync.set( obj, function(){
-					console.log( "updated cloud files with: ", obj, this, arguments);
-				});
-			});
-		}
-
-		for (var y in window.localStorage) {
-			console.log("checking", y);
-			obj[y] = window.localStorage.getItem(y);
-			if (y.indexOf(SOL_CACHE_FILE_PREFIX) !== 0) continue;
-			count++;
-			check(y);
-		}
 
+	// ------------------ query params (hash) ----------------
 
+	function syncQueryParams() {
+	  $('#optimize').attr( 'checked', (queryParams.get().optimize == "true") );
 	}
 
-	window.syncStorage = syncStorage;
-	syncStorage();
-
+	window.onhashchange = syncQueryParams;
+	syncQueryParams();
 
 
-	// ----------------- editor ----------------------
+	// ------------------ gist load ----------------
 
-	var SOL_CACHE_FILE_PREFIX = 'sol-cache-file-';
-	var SOL_CACHE_UNTITLED = SOL_CACHE_FILE_PREFIX + 'Untitled';
-	var SOL_CACHE_FILE = null;
-
-	var editor = ace.edit("input");
-	var sessions = {};
-
-	var Range = ace.acequire('ace/range').Range;
-	var errMarkerId = null;
-
-	var untitledCount = '';
-	if (!getFiles().length || window.localStorage['sol-cache']) {
-		if(loadingFromGist) return;
-		// Backwards-compatibility
-		while (window.localStorage[SOL_CACHE_UNTITLED + untitledCount])
-			untitledCount = (untitledCount - 0) + 1;
-		SOL_CACHE_FILE = SOL_CACHE_UNTITLED + untitledCount;
-		window.localStorage[SOL_CACHE_FILE] = window.localStorage['sol-cache'] || BALLOT_EXAMPLE;
-		window.localStorage.removeItem('sol-cache');
-	}
+	var loadingFromGist = gistHandler.handleLoad(function(gistId) {
+		$.ajax({
+	    url: 'https://api.github.com/gists/'+gistId,
+	    jsonp: 'callback',
+	    dataType: 'jsonp',
+	    success: function(response) {
+				if (response.data) {
+					if (!response.data.files) {
+						alert( "Gist load error: " + response.data.message );
+						return;
+					}
+					loadFiles(response.data.files);
+				}
+	    }
+    });
+	});
 
-	SOL_CACHE_FILE = getFiles()[0];
 
+	// ----------------- storage --------------------
 
-	var files = getFiles();
-	for (var x in files) {
-		sessions[files[x]] = newEditorSession(files[x]);
-	}
+	var storageHandler = new StorageHandler(updateFiles);
+	window.syncStorage = storageHandler.sync;
```

Comment:
I have no clue what `window.syncStorage` does. I'm just trying to preserve behavior since this is a refactoring.